satpy/scene.py

```diff
@@ -135,16 +135,15 @@ def sensor_names(self) -> set[str]:
         Sensor information is collected from data contained in the Scene
         whether loaded from a reader or generated as a composite with
         :meth:`load` or added manually using ``scn["name"] = data_arr``).
-        If no data is currently contained in the Scene or no sensor
-        information is found in the data metadata, loaded readers will be
-        consulted for sensor information.
+        Sensor information is also collected from any loaded readers.
+        In some rare cases this may mean that the reader includes sensor
+        information for data that isn't actually loaded or even available.
```

Comment:
How can this happen?

Reply:
Most likely bad configuration by the reader developer. Like if the file handler doesn't look at what file it is and just blindly says all possible sensors. Or if the file handler have a sensor names property, I think the reader defaults to the sensors listed in the YAML.